oc_cdtapi/RundeckAPI.py

```diff
@@ -0,0 +1,599 @@
+        """
+        self._logger.info(" ".join([
+            "Upating project SCM configuration:",
+            f"project: [{project}], integration: [{integration}], plugin_type: [{plugin_type}]",
+            f"type of definition: [{type(definition)}]"]))
+
+        # checking arguments
+        if not definition:
+            raise ValueError("SCM definition is mandatory")
+
+        if not project:
+            raise ValueError("Project name is mandatory")
+
+        if integration not in ["import", "export"]:
+            raise ValueError(f"Unsupported SCM integration type: [{integration}]")
+
+        if not plugin_type:
+            raise ValueError("Plugin type is mandatory")
+        
+        definition = self._to_dict(definition)
+
+        # configuration may be outside of a dict
+        if not definition.get("config"):
+            definition = {"config": definition}
+
+        self._logger.log(5, f"Final definition is: [{definition}]")
+
+        _req = ["project", project, "scm", integration, "plugin", plugin_type, "setup"]
+
+        return self.post(_req, headers=self.headers, cookies=self.cookies, data=json.dumps(definition)).json()
+
+    def scm__enable(self, project, integration, plugin_type, enable=False):
+        """
+        Turn on/off SCM for a project
+        :param str project: project name
+        :param str integration: one of: "import", "export"
+        :param str plugin_type: plugin to configure
+        :param bool enable: disable if set to False, enable otherwise
+        :return dict:
+        """
+        self._logger.info(
+                " ".join([
+                    'Enable' if enable else 'Disable',
+                    f"SCM for project [{project}], integration [{integration}], plugin_type [{plugin_type}]"]))
+
+        # checking arguments
+        if not project:
+            raise ValueError("Project name is mandatory")
+
+        if integration not in ["import", "export"]:
+            raise ValueError(f"Unsupported SCM integration type: [{integration}]")
+
+        if not plugin_type:
+            raise ValueError("Plugin type is mandatory")
+        
+        _req = ["project", project, "scm", integration, "plugin", plugin_type, "enable" if enable else "disable"]
+
+        return self.post(_req, headers=self.headers, cookies=self.cookies).json()
+
+    def scm__get_action_inputs(self, project, integration, action):
+        """
+        Get items needed to be performed actions on
+        :param str project: project name
+        :param str integration: one of: "import", "export"
+        :param str action: action to perform
+        :return dict:
+        """
+        self._logger.info(
+                f"Get SCM actions input list for project [{project}], integration [{integration}], action [{action}]")
+
+        # checking arguments
+        if not project:
+            raise ValueError("Project name is mandatory")
+
+        if integration not in ["import", "export"]:
```

Comment:
this check is performed a number of times. if list of valid values is subject to change in the future, I'd suggest to set it up in \_\_init\_\_ for further reference.